skrf/mathFunctions.py

```diff
@@ -1258,3 +1258,58 @@ def is_positive_semidefinite(mat: npy.ndarray, tol: float = ALMOST_ZERO) -> bool
     except npy.linalg.LinAlgError:
         return False
     return npy.all(v > -tol)
+
+def rsolve(A: npy.ndarray, B: npy.ndarray) -> npy.ndarray:
+    r"""Solves x @ A = B.
+
+    Calls numpy.linalg.solve with transposed matrices.
+
+    Input should have dimension of similar to (nfreqs, nports, nports).
+
+    Parameters
+    ----------
+    A : npy.ndarray
+    B : npy.ndarray
+
+    Returns
+    -------
+    x : npy.ndarray
+    """
+    return npy.transpose(npy.linalg.solve(npy.transpose(A, (0, 2, 1)).conj(),
+            npy.transpose(B, (0, 2, 1)).conj()), (0, 2, 1)).conj()
```

Comment:
Actually, this should work as well (tested with NumPy 1.20)
```suggestion
    return B @ npy.linalg.inv(A)
```

Reply:
The point of this function is to avoid calculating the inverse because calculating the inverse and then multiplying the matrices is numerically more inaccurate than directly solving the linear system. In Z-parameter calculations some of the numbers can be very big and we want to keep as much precision as possible.

Reply:
OK, good to know. Could please add such a comment in the function? In case somewhere in the future would think like me that it could be simplified.

Reply:
ok, already done. Perfect.